src/js/background/identityState.js

```diff
@@ -54,9 +106,57 @@ const identityState = {
     return this.storageArea.set(cookieStoreId, containerState);
   },
 
+  async updateUUID(cookieStoreId, uuid) {
+    if (cookieStoreId && uuid) {
+      const containerState = await this.storageArea.get(cookieStoreId);
+      containerState.macAddonUUID = uuid;
+      await this.storageArea.set(cookieStoreId, containerState);
+      return uuid;
+    } 
+    throw new Error ("cookieStoreId or uuid missing");
+  },
+
+  async addUUID(cookieStoreId) {
+    await this.storageArea.get(cookieStoreId);
+  },
+
+  async lookupMACaddonUUID(cookieStoreId) {
+    // This stays a lookup, because if the cookieStoreId doesn't 
+    // exist, this.get() will create it, which is not what we want.
+    const cookieStoreIdKey = cookieStoreId.includes("firefox-container-") ? 
+      cookieStoreId : "firefox-container-" + cookieStoreId;
+    const macConfigs = await this.storageArea.area.get();
+    for(const configKey of Object.keys(macConfigs)) {
+      if (configKey === "identitiesState@@_" + cookieStoreIdKey) {
```

Comment:
Can all the references to this key call into a reference?

But this one should be:
```suggestion
      if (configKey === this.getContainerStoreKey(cookieStoreIdKey)) {
```